### PR TITLE
Update release notes for docs changes

### PR DESCRIPTION
**Pull request by stichbury**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
Added some notes about what has changed in docs since the last release, and noted the community contributors that made some of those changes.

## Development notes
n/a

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -20,6 +20,21 @@
...
+* Moved documentation for the `standalone-datacatalog` starter into its [README file](https://github.com/kedro-org/kedro-starters/tree/main/standalone-datacatalog).
+* Added new documentation about deploying a Kedro project with Amazon EMR.
+* Added new documentation about how to publish a Kedro-Viz project to make it shareable.
+* New TSC members added to the page and the organisation of each member is also now listed.
+* Plus some minor bug fixes and changes across the documentation.
+
+## Community contributions
+Many thanks to the following Kedroids for contributing PRs to this release:
+
+* [Laíza Milena Scheid Parizotto](https://github.com/laizaparizotto)
```
Comment (ankatiyar) [inline RELEASE.md:33]: @stichbury There's another Community contributions section below (after the deprecation notices) with some more people, maybe we should combine them?

**Review by merelcht — APPROVED** [review]
Thanks for making the notes ready for release @stichbury ⭐ 😊

File: RELEASE.md [hunk 2]
```diff
@@ -31,6 +38,10 @@ Many thanks to the following Kedroids for contributing PRs to this release:
 
 * [Jason Hite](https://github.com/jasonmhite)
 * [IngerMathilde](https://github.com/IngerMathilde)
+* [Laíza Milena Scheid Parizotto](https://github.com/laizaparizotto)
+* [Richard](https://github.com/CF-FHB-X)
+* [flpvvvv](https://github.com/flpvvvv)
+* [qheuristics](https://github.com/qheuristics)
```
Comment (ankatiyar) [inline RELEASE.md:44]: There's also another contribution from @/miguel-ortiz-marin in this release

Comment (stichbury) [reply RELEASE.md:44]: Thanks, yes, there are probably more. I only looked at the docs subfolder for third party contributors and added them (if you recall, they were in a separate section of the release notes). So this is not a comprehensive list of community contributors but it _is_ comprehensive for docs (up to today at least 17/10)

**Review by ankatiyar — APPROVED** [review]
Thanks @stichbury! 😄

File: RELEASE.md (hunk 2 above)
Comment (ankatiyar) [reply RELEASE.md:44]: Yep, that makes sense, I thought it'd be easier to add it in this PR since there was only one more.

Comment (merelcht) [reply RELEASE.md:45]: I'll do another check for contributors on the release PR!

**Pull request merged**